File: scripts/test_solutions/271_encode_decode_strings.js
var encode = function(strs) {
    return strs.map(s => s.length + '#' + s).join('');
};
var decode = function(s) {
    const result = [];
    let i = 0;
    while (i < s.length) {
        let j = i;
        while (s[j] !== '#') j++;
        const len = parseInt(s.substring(i, j));
        result.push(s.substring(j + 1, j + 1 + len));
        i = j + 1 + len;
    }
    return result;
};

File: scripts/test_solutions/271_encode_decode_strings.cpp
class Solution {
public:
    string encode(vector<string>& strs) {
        string result;
        for (const string& s : strs) {
            result += to_string(s.length()) + "#" + s;
        }
        return result;
    }

    vector<string> decode(string s) {
        vector<string> result;
        int i = 0;
        while (i < s.length()) {
            int j = i;
            while (s[j] != '#') {
                j++;
            }
            int length = stoi(s.substr(i, j - i));
            result.push_back(s.substr(j + 1, length));
            i = j + 1 + length;
        }
        return result;
    }
};
